packages/preferences/src/store/selectors.js

```diff
@@ -1,3 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+
+const withDeprecatedKeys = ( originalGet ) => ( state, scope, name ) => {
+	const settingsToMoveToCore = [
+		'allowRightClickOverrides',
+		'distractionFree',
+		'editorMode',
+		'fixedToolbar',
+		'focusMode',
+		'hiddenBlockTypes',
+		'inactivePanels',
+		'keepCaretInsideBlock',
+		'mostUsedBlocks',
+		'openPanels',
+		'showBlockBreadcrumbs',
+		'showIconLabels',
+		'showListViewByDefault',
+	];
+
+	if (
+		settingsToMoveToCore.includes( name ) &&
+		[ 'core/edit-post', 'core/edit-site' ].includes( scope )
+	) {
+		deprecated(
+			`wp.data.select( 'core/preferences' ).get( '${ scope }', '${ name }' )`,
+			{
+				since: '6.5',
+				alternative: `wp.data.select( 'core/preferences' ).get( 'core', '${ name }' )`,
+			}
+		);
+
+		const value = originalGet( state, 'core', name );
+
+		// Hotfix for 17.5. Some of the preferences in the list above haven't been
+		// migrated to core in 17.5 (i.e: `editorMode`, https://github.com/WordPress/gutenberg/pull/57642))
+		// so we should fallback to the passed scope to avoid unexpected `undefined` values.
+		if ( value === undefined ) {
+			return originalGet( state, scope, name );
+		}
+
+		return originalGet( state, 'core', name );
+	}
+
+	return originalGet( state, scope, name );
```

Comment:
Hmm, I should have avoided the duplicate call here by returning `value` if it's not `undefined`. Anyway, it's not a huge deal since the `trunk` version will not need this piece of code, and this will be deprecated in a week. I'll work on a follow-up with test fixes and will include this as a minor improvement.